sdks/python/apache_beam/typehints/typed_pipeline_test.py

```diff
@@ -221,9 +221,46 @@ def repeat(s, *times):
     result = ['a', 'bb', 'c'] | beam.Map(repeat, 3)
     self.assertEqual(['aaa', 'bbbbbb', 'ccc'], sorted(result))
 
-  # TODO(robertwb): Support partially defined varargs.
-  # with self.assertRaises(typehints.TypeCheckError):
-  #   ['a', 'bb', 'c'] | beam.Map(repeat, 'z')
+    if sys.version_info >= (3,):
+      with self.assertRaisesRegexp(
+          typehints.TypeCheckError,
+          r'requires Tuple\[int, ...\] but got Tuple\[str, ...\]'):
+        ['a', 'bb', 'c'] | beam.Map(repeat, 'z')
+
+  def test_var_positional_only_side_input_hint(self):
+    # Test that a lambda that accepts only a VAR_POSITIONAL can accept
+    # side-inputs.
+    result = (['a', 'b', 'c']
+              | beam.Map(lambda *_: 'a', 5).with_input_types(str, int))
```

Comment:
Let's check that it actually gets the values right, e.g. do `lambda *args: args`. (Similarly below.)